### ISSUE TITLE
segfault with kernel 4.17.12

### ISSUE THREAD

**Issue by mouseMD2017** [camunda/camunda-modeler#869]
On arch linux, kernel 4.17.12 (x64), camunda modeler crashes on start:
`./camunda-modeler ->`

syslog:
`Aug 07 15:49:34 kernel: camunda-modeler[3340]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007fffc9efa768 error 15 in camunda-modeler[200000+1602000]
Aug 07 15:49:34 systemd-coredump[3342]: Process 3340 (camunda-modeler) of user 1000 dumped core.
Stack trace of thread 3340:
#0  0x0000000000a4fa30 n/a (/home/myuser/BPMN/camunda-modeler/camunda-modeler)`

Modeler version is `camunda-modeler-1.16.2-linux-x64.`

**Comment by nikku** [comment]
After a little bit of investigation I conclude that this is not an issue with the kernel version but an incompatibility introduced with the latest `glibc`.

I was able to fix the issue by downgrading the library to the previous version:

```
$ downgrade lib32-glibc glibc
....
warning: downgrading package lib32-glibc (2.28-1 => 2.27-3)
warning: downgrading package glibc (2.28-1 => 2.27-3)
...
```

Please do not report this to the Electron project. Instead [downgrade your `glibc`](https://wiki.archlinux.org/index.php/Downgrading_package) and wait for stuff to be fixed in the library.

**Comment by eli-schwartz** [comment]
Please don't report it to the Electron project? Why not, since it is a bug in electron which generates invalid code using an outdated `lld`?

https://github.com/electron/electron/pull/13988

In the meantime, consider using our system electron package, 1) which works fine, 2) is the superior solution due to not vendoring dozens of copies of electron all over the system...

**Comment by nikku** [comment]
Thanks for the heads-up. We will update this app with a compatible version of electron, once the linked fixes are released. Until then users experiencing this issue may fall back to Camunda Modeler v1.14 or use [this workaround](https://github.com/camunda/camunda-modeler/issues/869#issuecomment-411084797).

**Comment by nikku** [comment]
> Please don't report it to the Electron project? Why not, since it is a bug in electron which generates invalid code using an outdated lld?

I'm not here to blame either project for causing this issue. As such, I don't encourage people to create bug tickets all over the place before I'm confident of the root cause of an issue.

I've updated my [previous comment](https://github.com/camunda/camunda-modeler/issues/869#issuecomment-411084797) accordingly.

**Comment by nikku** [comment]
Discussion is ongoing how things are going to be addressed in Electron, cf. https://github.com/electron/electron/pull/13988#pullrequestreview-145371196.

**Comment by nikku** [comment]
Arch reverted the offending changes in glibc for now which means the app should run on Arch Linux again.

**Comment by nikku** [comment]
The longer term fix in Electron is currently pending release.

**Comment by nikku** [comment]
Fixed with `v2.0.0` of the Camunda Modeler.
